File: Corgi.Backend/Dtos/v1/GetTemplateDtoV1.cs
using Corgi.Backend.Models;

namespace Corgi.Backend.Dtos.v1
{
    public class GetTemplateDtoV1
    {
        public Guid Id { get; set; }
        public string Name { get; set; }
        public List<GetTemplateFieldDtoV1> Fields { get; set; } = [];
        public DateTime CreatedAt { get; set; }
        public DateTime LastModified { get; set; }
    }
}

File: Corgi.Frontend/src/app/get-template-dto-v1.ts
import { GetTemplateFieldDtoV1 } from "./get-template-field-dto-v1";

export class GetTemplateDtoV1 {
    constructor(
        public id: string
        , public name: string
        , public fields: GetTemplateFieldDtoV1[]
        , public createdAt: string
        , public lastModified: string
    ) {

    }
}
